pkg/openstack/actuator.go

```diff
@@ -136,10 +140,30 @@ func (a *OpenStackActuator) sync(ctx context.Context, cr *minterv1.CredentialsRe
 	return nil
 }
 
-func (a *OpenStackActuator) prepareCloudsData(clouds string) (string, error) {
-	cloud, err := gophercloudutils.
+func (a *OpenStackActuator) prepareCloudsData(content string) (string, error) {
+	var clouds clientconfig.Clouds
 
-	return clouds, nil
+	err := yaml.Unmarshal([]byte(content), &clouds)
+	if err != nil {
+		return "", err
+	}
+
+	if clouds.Clouds[rootOpenStackCloudName].CACertFile == caCertFile || clouds.Clouds[rootOpenStackCloudName].CACertFile == "" {
+		// Don't have to modify anything
+		return content, nil
+	}
+
+	// CACertFile has an incorrect value, we need to replace it with the canonical one
+	newCloud := clouds.Clouds[rootOpenStackCloudName]
+	newCloud.CACertFile = caCertFile
+	clouds.Clouds[rootOpenStackCloudName] = newCloud
```

Comment:
```
clouds.Clouds[rootOpenStackCloudName].CACertFile = caCertFile
```

Reply:
thank you for the suggestion but unfortunately your code won't compile with the next error:

> cannot assign to struct field clouds.Clouds[rootOpenStackCloudName].CACertFile in mapcompilerUnaddressableFieldAssign